src/encoding/armor.js

```diff
@@ -247,6 +247,9 @@ function splitChecksum(text) {
   if (lastEquals >= 0) {
     body = text.slice(0, lastEquals);
     checksum = text.slice(lastEquals + 1);
+    if (checksum.substr(0, 6) === '\n-----') {
```

Comment:
Check this out, add a log
```js
  if (lastEquals >= 0) {
    body = text.slice(0, lastEquals);
    checksum = text.slice(lastEquals + 1);
    console.log(checksum);
    if (checksum.substr(0, 6) === '\n-----') {
      checksum = ''; // missing armor checksum
    }
  }
```
then run `npm test`
```
              ✓ should fail for incorrect passphrase (308ms)
            encryptSessionKey, decryptSessionKey
h/aX
-----END PGP PUBLIC KEY BLOCK-----
lw5e
-----END PGP PRIVATE KEY BLOCK-----
              ✓ should encrypt with public key
h/aX
-----END PGP PUBLIC KEY BLOCK-----
lw5e
-----END PGP PRIVATE KEY BLOCK-----
              ✓ should encrypt with password
h/aX
-----END PGP PUBLIC KEY BLOCK-----
lw5e
-----END PGP PRIVATE KEY BLOCK-----
0WSm

0WSm
```
The ending header is still in there. You wouldn't know, because it gets cut later with `checksum = checksum.substr(0, 4);`

It didn't use to be this way, I think older versions did not let the ending header slip through to the next step.

Personally, I think `splitChecksum`'s `checksum` should either return a 4 char string or a null, but I was trying to keep the surgery at a minimum :)

Anyway, that's why this approach. Should I try to clean it up?